src/common/types/types.cpp

```diff
@@ -605,6 +605,19 @@ bool LogicalTypeUtils::isNumerical(const LogicalType& dataType) {
     }
 }
 
+bool LogicalTypeUtils::isPrimitive(const LogicalType& dataType) {
```

Comment:
Agree on the part of "be more prone to be duplicated without knowing the existence of this". But I'm not sure if another function needs this later, it will end up with the same definition of `isPrimitive`.
`isNested` looks clearer to me. I'm fine to keep it inside `LogicalTypeUtils`.